supervisor/homeassistant/websocket.py

```diff
@@ -24,29 +25,34 @@ def __init__(
         self, ha_version: AwesomeVersion, client: aiohttp.ClientWebSocketResponse
     ):
         """Initialise the WS client."""
-        self.ha_version = ha_version
-        self.client = client
-        self.message_id = 0
+        self.ha_version: AwesomeVersion = ha_version
+        self.client: aiohttp.ClientWebSocketResponse = client
+        self.message_id: int = 0
+        self._lock: asyncio.Lock = asyncio.Lock()
 
     async def async_send_command(self, message: Dict[str, Any]):
         """Send a websocket command."""
-        self.message_id += 1
-        message["id"] = self.message_id
+        while self._lock.locked():
+            await asyncio.sleep(0)
```

Comment:
Why not wait on the lock itself? we can also create it on the job, `@Job(limit=ExecutionLimit.SingleWait)` would be pretty easy to add into Job decorator and we can reuse it

Reply:
Before you close the PR: https://github.com/home-assistant/supervisor/pull/2612 - You can use that

Reply:
We can not and should not use Job here, it's a helper.
In the future all jobs will also use this helper to publish their updates, at that point we will end in a loop